ui/src/warp_runner/manager/commands/constellation_commands.rs

```diff
@@ -319,10 +329,76 @@ fn rename_if_duplicate(
     new_file_name
 }
 
+async fn set_thumbnail_if_file_is_video(
+    warp_storage: &warp_storage,
+    filename_to_save: String,
+    file_path: PathBuf,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let video_formats = VIDEO_FILE_EXTENSIONS.to_vec();
+
+    let file_extension = std::path::Path::new(&filename_to_save)
+        .extension()
+        .and_then(OsStr::to_str)
+        .map(|s| format!(".{s}"))
+        .unwrap_or_default();
+
+    if !video_formats.iter().any(|f| f == &file_extension) {
+        log::warn!("It is not a video file!");
+        return Err(Box::from(Error::InvalidDataType));
+    };
+
+    let item = warp_storage
+        .current_directory()?
+        .get_item(&filename_to_save)?;
+
+    let file_stem = file_path
+        .file_stem()
+        .and_then(OsStr::to_str)
+        .map(str::to_string)
+        .unwrap_or_default();
+
+    let temp_dir = TempDir::new()?;
+
+    let temp_path = temp_dir.path().join(file_stem);
+
+    let output = Command::new("ffmpeg")
```

Comment:
Since this is done in a async function, you would want to use the `Command` from tokio to prevent blocking the current thread.